debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/connection/pgoutput/PgOutputReplicationMessage.java

```diff
@@ -0,0 +1,321 @@
+import org.postgresql.util.PGInterval;
+import org.postgresql.util.PGmoney;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import io.debezium.connector.postgresql.PostgresType;
+import io.debezium.connector.postgresql.RecordsStreamProducer.PgConnectionSupplier;
+import io.debezium.connector.postgresql.connection.ReplicationMessage;
+import io.debezium.connector.postgresql.connection.wal2json.DateTimeFormat;
+import io.debezium.data.SpecialValueDecimal;
+import io.debezium.time.Conversions;
+import io.debezium.util.Strings;
+
+/**
+ * @author Gunnar Morling
+ * @author Chris Cranford
+ */
+public class PgOutputReplicationMessage implements ReplicationMessage {
+    private static final Logger LOGGER = LoggerFactory.getLogger(PgOutputReplicationMessage.class);
+
+    private Operation op;
+    private Instant commitTimestamp;
+    private long transactionId;
+    private String table;
+    private List<Column> oldColumns;
+    private List<Column> newColumns;
+
+    public PgOutputReplicationMessage(Operation op, String table, Instant commitTimestamp, long transactionId, List<Column> oldColumns, List<Column> newColumns) {
+        this.op = op;
+        this.commitTimestamp = commitTimestamp;
+        this.transactionId = transactionId;
+        this.table = table;
+        this.oldColumns = oldColumns;
+        this.newColumns = newColumns;
+    }
+
+    @Override
+    public Operation getOperation() {
+        return op;
+    }
+
+    @Override
+    public Instant getCommitTime() {
+        return commitTimestamp;
+    }
+
+    @Override
+    public long getTransactionId() {
+        return transactionId;
+    }
+
+    @Override
+    public String getTable() {
+        return table;
+    }
+
+    @Override
+    public List<Column> getOldTupleList() {
+        return oldColumns;
+    }
+
+    @Override
+    public List<Column> getNewTupleList() {
+        return newColumns;
+    }
+
+    @Override
+    public boolean hasTypeMetadata() {
+        return true;
+    }
+
+    @Override
+    public boolean isLastEventForLsn() {
+        return true;
+    }
+
+    @Override
+    public boolean shouldSchemaBeSynchronized() {
+        return false;
+    }
+
+    /**
+     * Converts the value (string representation) coming from PgOutput plugin to
+     * a Java value based on the type of the column from the message.  This value will be converted later on if necessary by the
+     * connector's value converter to match whatever the Connect schema type expects.
+     *
+     * Note that the logic here is tightly coupled on the pgoutput plugin logic which writes the actual value.
+     *
+     * @return the value; may be null
+     */
+    public static Object getValue(String columnName, PostgresType type, String fullType, String rawValue, final PgConnectionSupplier connection, boolean includeUnknownDataTypes) {
+        // todo: DBZ-766 this is a slightly modified version from Wal2JsonReplicationMessage#getValue
```

Comment:
How much both wal2json and pgoutput method differs? If only slightly then it makes sense to consolidate it now othrwise it will make a problem in next update of types

Reply:
I'll have to see if there is any good way to consolidate this.  The main difference is the fact that the wal2json plugin operates on a `Value` wrapper object where pgoutput operates directly on strings and I don't believe it makes sense to try and wrap the pgoutput string in a `Value`.

Reply:
I added a new interface `ReplicationMessage.ColumnValue<T>`.  Let me know if that seems acceptable to you as I was able to refactor a vast majority of the same logic into an abstract implementation and override the pertinent parts per decoder.

Reply:
@Naros Excellent work!